### PR TITLE
Fix very short spinners being impossible to complete

### LINKED ISSUE #5826: Spinners too short are in-completeable
**Describe the missing feature:** 
If a spinner is too short, it will not be able to be completed, making a map un-fcable, even with auto
an example of a map that has this issue is : https://osu.ppy.sh/beatmapsets/600897#osu/1269591
**Proposal designs of the feature:** 
in stable if a spinner is too short to complete the spinner will auto complete itself, it should do the same in lazer, if a spinner can't be completed normally it should just complete itself and give a 300

### PR DESCRIPTION
**Pull request by peppy**
- Closes #5826

If the calculated spin requirement is lower than 1, implicit completion is expected.

Can be tested on https://osu.ppy.sh/beatmapsets/132#osu/296.

### CODE REVIEW

**Comment by frenzibyte** [comment]
Sounds like it closes https://github.com/ppy/osu/issues/5826.

File: osu.Game.Rulesets.Osu.Tests/TestSceneSpinner.cs [hunk 1]
```diff
@@ -17,32 +17,54 @@ public class TestSceneSpinner : OsuSkinnableTestScene
...
-        public TestSceneSpinner()
+        private TestDrawableSpinner drawableSpinner;
+
+        [TestCase(false)]
+        [TestCase(true)]
+        public void TestVariousSpinners(bool autoplay)
         {
             AddStep("Miss Big", () => SetContents(() => testSingle(2)));
             AddStep("Miss Medium", () => SetContents(() => testSingle(5)));
             AddStep("Miss Small", () => SetContents(() => testSingle(7)));
```
Comment (bdach) [inline osu.Game.Rulesets.Osu.Tests/TestSceneSpinner.cs:28]: Seeing the `[TestCase]` attributes added on this method, did you mean to remove these and replace them with the `false` test case? As it stands these are covered doubly.

File: osu.Game.Rulesets.Osu.Tests/TestSceneSpinner.cs [hunk 2]
```diff
@@ -17,32 +17,54 @@ public class TestSceneSpinner : OsuSkinnableTestScene
...
+            AddStep("Very short spinner", () => SetContents(() => testSingle(5, autoplay, 200)));
+            AddUntilStep("Wait for completion", () => drawableSpinner.Result.HasResult);
+            AddUntilStep("Short spinner implicitly completes", () => drawableSpinner.Progress == 1);
         }
 
-        private Drawable testSingle(float circleSize, bool auto = false)
+        private Drawable testSingle(float circleSize, bool auto = false, double length = 3000)
         {
-            var spinner = new Spinner { StartTime = Time.Current + 2000, EndTime = Time.Current + 5000 };
+            var spinner = new Spinner { StartTime = Time.Current + 2000, EndTime = Time.Current + +2000 + length };
```
Comment (bdach) [inline osu.Game.Rulesets.Osu.Tests/TestSceneSpinner.cs:54]: Not sure what the unary `+` is for:

```suggestion
            var spinner = new Spinner { StartTime = Time.Current + 2000, EndTime = Time.Current + 2000 + length };
```

**Review by bdach — COMMENTED** [review]
looks good otherwise :+1:

File: osu.Game.Rulesets.Osu.Tests/TestSceneSpinner.cs (hunk 2 above)
Comment (peppy) [reply osu.Game.Rulesets.Osu.Tests/TestSceneSpinner.cs:54]: typo

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**